qutebrowser/browser/hints.py

```diff
@@ -355,6 +356,90 @@ def _set_style_properties(self, elem, label):
             label.setStyleProperty(k, v)
         self._set_style_position(elem, label)
 
+    def _get_element_main_rect(self, elem, show):
+        """Get the element geometry.
+
+        Args:
+            elem: The QWebElement we want to get the geometry.
+            show: Whether we want to position to show or not (click).
+
+        Return:
+            A QRect.
+        """
+        rect = QRect()
+        js_rects = elem.evaluateJavaScript('this.getClientRects()')
+        if js_rects:
+            max_width = 0
+            max_index = '0'
+            size = int(js_rects['length'])
+            if size > 0:
+                for i in range(size):
+                    index = str(i)
+                    width = js_rects[index]['width']
+                    if width > max_width:
+                        max_width = width
+                        max_index = index
+                        break
+                js_rect = js_rects[max_index]
+                rect = QRect(js_rect['left'], js_rect['top'], js_rect['width'],
+                             js_rect['height'])
+
+            if show:
+                body = elem.document().findFirst('body')
+                if(body.styleProperty('position', QWebElement.ComputedStyle) ==
+                        'relative'):
+                    body_rect = elem.evaluateJavaScript(
+                        'document.body.getBoundingClientRect()'
+                    )
+                    rect.translate(-body_rect['left'], -body_rect['top'])
+                    frame = elem.webFrame()
+                    rect.translate(-frame.scrollPosition())
```

Comment:
I don't see an issue - I just want to avoid more complexity if it can be avoided :wink:

I hope I'll get to writing some tests for hint positioning soon. I definitely have some corner cases in mind.